app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -10,141 +10,108 @@
-        if (!array_has($values, ['northstar_id', 'campaign_id', 'campaign_run_id'])) {
-            // If we have valid campaign values, use em! This also means that we do not have NS id
-            if (array_has($values, ['campaign_id', 'campaign_run_id']) && is_numeric($values['campaign_id']) && is_numeric($values['campaign_run_id'])) {
-                $finalValues = [
-                    'northstar_id' => null, // set the user to null so we force account creation when the code is not present.
-                    'campaign_id' => $values['campaign_id'],
-                    'campaign_run_id' => $values['campaign_run_id'],
-                    'source' => $this->source,
-                    'source_details' => null,
-                ];
+            elseif (str_contains($value, '=')) {
+                $value = explode('=', $value);
             }
 
-            // If we have NS id, use it! This also means that we do not have both campaign_id and campaign_run_id, so use the defaults
-            if (array_has($values, ['northstar_id'])) {
-                $finalValues = [
-                    'northstar_id' => $values['northstar_id'], // set the user to null so we force account creation when the code is not present.
-                    'campaign_id' => 8017,
-                    'campaign_run_id' => 8022,
-                    'source' => $this->source,
-                    'source_details' => null,
-                ];
+            $key = strtolower($value[0]);
+            // We expect 'user', but check for any variations/typos in any manually entered URLs.
+            if ($key === 'user' || $key === 'user_id' || $key === 'userid') {
+                $userId = $value[1];
             }
-        }
 
-        // If we were missing all the necessary values or if this is a referral, use all the defaults
-        if (empty($finalValues) || $referral) {
-            $finalValues = [
-                'northstar_id' => null, // set the user to null so we force account creation when the code is not present.
-                'campaign_id' => 8017,
-                'campaign_run_id' => 8022,
-                'source' => $this->source,
-                'source_details' => null,
-            ];
+            /**
+             * If referral parameter is set to true, the user parameter belongs to the referring
+             * user, not the user that should be associated with this voter registration record.
+             */ 
+            // We expect 'referral', but check for any typos in any manually entered URLs.
+            if (($key === 'referral' || $key === 'refferal') && str_to_boolean($value[1])) {
+                /**
+                 * Return null to force querying for existing user via this record email or mobile
+                 * upon import.
+                 */
+                return null;
+            }
         }
 
-        return $finalValues;
+        return isset($userId) ? $userId : null;
```

Comment:
The `isset()` here gives me a little pause (since as a reader, I have to then Cmd+F to see where it may or may not get set above). Can we instead order the conditionals so this reads "in order"? e.g.

```php
    // We expect 'referral', but check for any typos in any manually entered URLs.
    if (($key === 'referral' || $key === 'refferal') && str_to_boolean($value[1])) {
        return null;
    }

    // We expect 'user', but check for any variations/typos in any manually entered URLs.
    if ($key === 'user' || $key === 'user_id' || $key === 'userid') {
        return $value[1];
    }
}

return null;
```

Reply:
I didn't return the `userId` if found -- to handle an edge case where an editor may add the referral parameter after the user parameter, e.g. `puppetsloth.org?user=5547be89469c64ec7d8b518d&referral=true` -- In this case we want to return `null` for a user ID

Reply:
We could set `$userId` to `null` in the beginning of the function? I had that at first, replacing `$values` -- but then went this route

Reply:
Oh! I didn't notice this was inside a `foreach`! 🤦‍♂️ 

It might make sense to make a first pass to "read" the referral string into an array (e.g. `user=5547be89469c64ec7d8b518d&referral=true` into `['user' => '…', 'referral' => true]`, since then you can have more "straightforward" conditions without having to worry about ordering. But definitely not a big thing!

Reply:
Sounds good - we can revisit when ready to send over an `action_id` instead of `campaign_id`